### PR TITLE
feat(meroxa): Add Meroxa spec

### PR DESCRIPTION
**Pull request by raulb**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

This pull request introduces a new spec for the [Meroxa](https://meroxa.com/) CLI: https://github.com/meroxa/cli.

**What is the current behavior? (You can also link to an open issue here)**

Meroxa CLI users can't benefit from the awesomeness of Fig. 😿

**What is the new behavior (if this is a feature change)?**

Meroxa CLI users that also have Fig installed can start using the great autocomplete feature! 💃 

**Additional info:**

Spec [was automatically generated](https://github.com/meroxa/cli-autocomplete/pull/1) using the GitHub action [withfig/push-to-fig-autocomplete-action](https://github.com/withfig/push-to-fig-autocomplete-action) however it seems that documentation needs to be updated. This is the GitHub action I had to [use instead](https://github.com/meroxa/cli/pull/403/files#diff-dc5b570ffcac9642dec0a589a111dd7afa53251f9b30e0a72958687a00b3954aR28). 

🔈 Reported in your Discord channel: https://discord.com/channels/837809111248535583/839904703214649345/1009376046279311460.

I have some other feedback on the documentation, but I'll send out via email to hello@fig.io during the day. I simply wanted to get things going 🍻 

Let me know if you have any question! I'll be around in your Discord channel if that helps.

### CODE REVIEW

File: src/meroxa.ts [hunk 1]
```diff
@@ -0,0 +1,664 @@
...
+      name: ["app", "apps"],
+      description: "Manage Turbine Data Applications (Beta)",
+      subcommands: [
+        {
+          name: "deploy",
+          description: "Deploy a Turbine Data Application (Beta)",
+          options: [
+            {
+              name: "--docker-hub-access-token",
+              description: "",
```
Comment (SeparateRecords) [inline src/meroxa.ts:16]: Would you mind adding a description here?

File: src/meroxa.ts [hunk 2]
```diff
@@ -0,0 +1,664 @@
...
+              hidden: true,
+              args: { name: "docker-hub-access-token" },
+            },
+            {
+              name: "--docker-hub-username",
+              description: "",
```
Comment (SeparateRecords) [inline src/meroxa.ts:22]: Same goes for the other options with no description (they still show up on https://fig.io/manual)

File: src/meroxa.ts [hunk 3]
```diff
@@ -0,0 +1,664 @@
...
+          ],
+        },
+        {
+          name: "run",
+          description: "Execute a Turbine Data Application locally (Beta)",
+          options: [
+            {
+              name: ["--lang", "-l"],
+              description: "Language to use (go | js)",
+              args: { name: "lang" },
```
Comment (SeparateRecords) [inline src/meroxa.ts:97]:
```suggestion
              args: { name: "lang", suggestions: ["go", "js"] },
```

File: src/meroxa.ts [hunk 4]
```diff
@@ -0,0 +1,664 @@
...
+            },
+            {
+              name: "--path",
+              description: "Path of application to run",
+              args: { name: "path" },
```
Comment (SeparateRecords) [inline src/meroxa.ts:102]:
```suggestion
              args: { name: "path", template: "filepaths" },
```

File: src/meroxa.ts [hunk 5]
```diff
@@ -0,0 +1,664 @@
...
+          name: ["log", "logs"],
+          description:
+            "View relevant logs to the state of the given Turbine Data Application (Beta)",
+        },
+        {
+          name: ["rm", "delete", "remove"],
+          description: "Removes a Turbine Data Application (Beta)",
+          options: [
+            { name: ["--force", "-f"], description: "Skip confirmation" },
+            { name: "--yolo", description: "Skip confirmation", hidden: true },
```
Comment (SeparateRecords) [inline src/meroxa.ts:87]: Love this!! 😄

File: src/meroxa.ts [hunk 6]
```diff
@@ -0,0 +1,664 @@
...
+            {
+              name: "--mod-vendor",
+              description:
+                "Whether to download modules to vendor or globally while initializing a Go application",
+            },
+            {
+              name: "--path",
+              description:
+                "Path where application will be initialized (current directory as default)",
+              args: { name: "path" },
```
Comment (SeparateRecords) [inline src/meroxa.ts:58]:
```suggestion
              args: { name: "path", template: "folders" },
```

File: src/meroxa.ts [hunk 7]
```diff
@@ -0,0 +1,664 @@
...
+          description: "Describe a Turbine Data Application (Beta)",
+        },
+        {
+          name: "init",
+          description: "Initialize a Turbine Data Application (Beta)",
+          options: [
+            {
+              name: ["--lang", "-l"],
+              description: "Language to use (js|go|py) (required)",
+              args: { name: "lang" },
```
Comment (SeparateRecords) [inline src/meroxa.ts:46]:
```suggestion
              args: { name: "lang", suggestions: ["js", "go", "py"] },
```

File: src/meroxa.ts [hunk 8]
```diff
@@ -0,0 +1,664 @@
...
+      description: "Manage endpoints on Meroxa",
+      subcommands: [
+        {
+          name: "create",
+          description: "Create an endpoint",
+          options: [
+            {
+              name: ["--protocol", "-p"],
+              description: "Protocol, value can be http or grpc (required)",
+              args: { name: "protocol" },
```
Comment (SeparateRecords) [inline src/meroxa.ts:165]:
```suggestion
              args: { name: "protocol", suggestions: ["http", "grpc"] },
```

File: src/meroxa.ts [hunk 9]
```diff
@@ -0,0 +1,664 @@
...
+        {
+          name: "update",
+          description: "Update an environment",
+          options: [
+            {
+              name: ["--config", "-c"],
+              description:
+                "Updated environment configuration based on type and provider (e.g.: --config aws_access_key_id=my_access_key --config aws_access_secret=my_access_secret)",
+              args: { name: "config" },
+            },
```
Comment (SeparateRecords) [inline src/meroxa.ts:258]: Looks like this should have `isRepeatable: true`

File: src/meroxa.ts [hunk 10]
```diff
@@ -0,0 +1,664 @@
...
+      description: "API url",
+      isPersistent: true,
+      hidden: true,
+      args: { name: "api-url" },
+    },
+    {
+      name: "--cli-config-file",
+      description: "Meroxa configuration file",
+      isPersistent: true,
+      args: { name: "cli-config-file" },
```
Comment (SeparateRecords) [inline src/meroxa.ts:647]:
```suggestion
      args: { name: "cli-config-file", template: "filepaths" },
```

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
Looks good! Just a little bit of polish that the integration didn't add, then it's ready to merge :)

**Comment by fedeci** [comment]
> Spec [was automatically generated](https://github.com/meroxa/cli-autocomplete/pull/1) using the GitHub action [withfig/push-to-fig-autocomplete-action](https://github.com/withfig/push-to-fig-autocomplete-action) however it seems that documentation needs to be updated. This is the GitHub action I had to [use instead](https://github.com/meroxa/cli/pull/403/files#diff-dc5b570ffcac9642dec0a589a111dd7afa53251f9b30e0a72958687a00b3954aR28).

Hey @raulb! Thanks for the spec, can I ask you what was wrong with the github action? I'll get it fixed asap!

**Comment by raulb** [comment]
> Hey @raulb! Thanks for the spec, can I ask you what was wrong with the github action? I'll get it fixed asap!

@fedeci Once using `withfig/push-to-fig-autocomplete-action` everything went smoothly! (except when I was confused about what should be in `spec-path` as I thought it was simply the path, and not including the actual `.ts` as well. My main issue came from the documentation part. Everything I read was referring to `withfig/publish-to-fig-autocomplete@v1` (example: https://fig.io/docs/guides/integrating/push-to-autocomplete-action), and I thought it was an issue on my end.

**Comment by fedeci** [comment]
Oh, that was the previous name of the action, I am going to fix it!

**Comment by raulb** [comment]
> Oh, that was the previous name of the action, I am going to fix it!

@fedeci thank you!

@SeparateRecords thank you for the review! Good catch on the `description` tag. Looks like that was something wrong in our end https://github.com/meroxa/cli/commit/22835871c60a8c5efa98828c7f8075fae21bd3ba. For `template` and `suggestions` I'm going to see if I can integrate that automatically so future releases don't mess up with that. I'll leave this PR open until I figure that out. Worst case scenario, I'll update the spec manually with those values until I find a more stable way todo this. Thank you again!

**Comment by SeparateRecords** [comment]
> I'm going to see if I can integrate that automatically so future releases don't mess up with that

Actually the push to autocomplete action will handle merging this automatically! No need to do anything on your end - you can add stuff manually once then you don't need to worry about it again. That's how the Fig spec works, and it's got quite a few generators!

File: src/meroxa.ts (hunk 9 above)
Comment (raulb) [reply src/meroxa.ts:258]: @SeparateRecords found the issue! https://github.com/withfig/autocomplete-tools/pull/73

**Comment by raulb** [comment]
> > I'm going to see if I can integrate that automatically so future releases don't mess up with that
> 
> Actually the push to autocomplete action will handle this automatically! No need to do anything on your end - you can add stuff manually once then you don't need to worry about it again. That's how the Fig spec works, and it's got quite a few generators!

@SeparateRecords maybe I'm missing something, but as far as I can tell, the GitHub action will generate a new spec based on the CLI structure so even though I added those fields manually on this pull-request now, a next update will show up as those being modified (in blank) so I would need to add them back manually again, right?

My current thinking is to try to use Cobra annotations to see if these could be generated as discussed with @mschrage in https://discord.com/channels/837809111248535583/852418354068324442/1009512317504983163.

Let me know what you think!

**Comment by fedeci** [comment]
Nope @raulb, that's correct for everything that could be generated automatically, but in the push-to-fig-autocomplete-action we run the merge tool with some [presets](https://github.com/withfig/autocomplete-tools/blob/main/merge/src/presets.ts) before creating the PR so we can keep generators and other manually set properties.
A preset for an integration contains all the properties that get generated by that integration, all the other properties are considered as manually set by the user and are retained.

**Comment by raulb** [comment]
> Nope @raulb, that's correct for everything that could be generated automatically, but in the push-to-fig-autocomplete-action we run the merge tool with some [presets](https://github.com/withfig/autocomplete-tools/blob/main/merge/src/presets.ts) before creating the PR so we can keep generators and other manually set properties. A preset for an integration contains all the properties that get generated by that integration, all the other properties are considered as manually set by the user and are retained.

Interesting! Alright! I just updated this pull-request under @SeparateRecords recommendations (co-authored you on your suggestions 🥂 ) so hopefully this will be ready to ship! 🤞 . In the meantime, I'll leave our own [pull-request](https://github.com/meroxa/cli/pull/403) ready for review for future releases.

**Review by SeparateRecords — APPROVED** [review]
LGTM, Fantastic job on this PR! Thank you so much for the contribution 🔥

**Comment by raulb** [comment]
@SeparateRecords thank you! I'm going to prepare [a changelog](https://docs.meroxa.com/changelog) announcing this. What's usually Fig's release cycle to make sure it's publicly available when we announce it?

Thank you again! I really appreciate the prompt reviews from y'all 🥂

**Comment by fedeci** [comment]
@raulb As you can see from the GH action it has already been released, however it may take a couple days to be available because of CDN caching.

**Comment by raulb** [comment]
@fedeci Make sense, grazie mile!

**Pull request merged**